### ISSUE TITLE
Is user configurable realm supported by HttpTransportSecurity?

### ISSUE THREAD

**Issue by tparvi** [dotnet/wcf#1992]
We have SAP that offers WCF interface. The following code written against full framework works as expected:

```
var address = new EndpointAddress(....);
var binding = new BasicHttpBinding(BasicHttpSecurityMode.TransportCredentialOnly);
binding.Security.Transport.ClientCredentialType = HttpClientCredentialType.Basic;
binding.Security.Transport.Realm = "SAP Web Application Server [LPQ]";

var client = new Client(binding, address);
client.ChannelFactory.Credentials.UserName.UserName = "user";
client.ChannelFactory.Credentials.UserName.Password = "pw";

var response = client.GetFoo(foo);
```

I'm trying to re-create this code in .NET Core and it is identical except that I cannot set the Realm property because System.ServiceModel.HttpTransportSecurity doesn't have it. So my call fail with "System Error" and no further details.

However I can see the Realm property in the System.Private.ServiceModel.HttpTransportSecurity class.

I'm using the v4.3.0 of the System.ServiceModel.Http and other nuget packages and targeting .NET Core 1.1.

**Comment by mconnew** [comment]
The Realm property is only used on the server side with WCF. By setting the realm, when an http client connects to the server, the authentication challenge will include the realm. This is generally used when a UI is needed to obtain credentials, e.g. with a web browser. This property has absolutely no effect on WCF clients and it's a mistake that it exists on the implementation class.

**Comment by mconnew** [comment]
@tparvi, does my answer resolve your issue? You should just be able to remove that line.

**Comment by tparvi** [comment]
@mconnew Yes, I think that explains everything. I'll have to see if there is some other issue that causes the "System Error" and I'll open up a new issue if needed. Thanks for the help!

**Comment by tparvi** [comment]
@iamxiaozhuang Yes we got it to work but I cannot remember how. However I looked at the code just now and it is same as what I posted here and I'm not setting the real property. I believe the issue was somewhere else. Maybe in the proxy classes that were generated? I'm not 100% sure but maybe we created the proxy classes in full framework and then just copy pasted them into .net core solution and fixed the compilation errors. If I would have to solve this issue now (as I have clearly forgotten how I fixed back then) I would create cmd line app with full framework and see if it works. Then generate the proxy classes on separate .net core cmd line app and compare them.
